compiler/rustc_query_impl/src/keys.rs

```diff
@@ -543,3 +544,19 @@ impl<'tcx> Key for (Ty<'tcx>, ty::ValTree<'tcx>) {
         DUMMY_SP
     }
 }
+
+impl Key for HirId {
+    #[inline(always)]
+    fn query_crate_is_local(&self) -> bool {
+        true
+    }
+
+    fn default_span(&self, tcx: TyCtxt<'_>) -> Span {
+        self.owner.default_span(tcx)
+    }
+
+    #[inline(always)]
+    fn key_as_def_id(&self) -> Option<DefId> {
+        Some(self.owner.to_def_id())
```

Comment:
are you sure this is correct? shouldn't we do `tcx.hir().local_def_id(self)` instead? ditto for `default_span`